dv/uvm/core_ibex/riscv_dv_extension/csr_description.yaml

```diff
@@ -288,25 +293,34 @@
 
 # TODO: Add support for WARL fields to CSR test generator allowing this CSR to
 # be tested.
-## MCAUSE
-#- csr: mcause
-#  description: >
-#    Indicates trap cause
-#  address: 0x342
-#  privilege_mode: M
-#  rv32:
-#    - field_name: Interrupt
-#      description: >
-#        Indicates if trap caused by interrupt
-#      type: WARL
-#      reset_val: 0
-#      msb: 31
-#      lsb: 31
-#    - field_name: Exception Code
-#      type: WLRL
-#      reset_val: 0
-#      msb: 4
-#      lsb: 0
+# MCAUSE
+- csr: mcause
+  description: >
+    Indicates trap cause
+  address: 0x342
+  privilege_mode: M
+  rv32:
+    - field_name: Interrupt info
+      description: >
+        Indicates if trap caused by interrupt and if that interrupt was internal
+        or external
+      type: WARL
```

Comment:
In the privileged spec section 3.1.15 it says that the exception code is from 30 to 0 and:
> The Exception Code is a WLRL field, so is only guaranteed to hold supported exception codes.

Where does this interrupt info come from? The interrupt field of mcause is only specified as being 1 bit wide in the spec.

Reply:
This is due to the Ibex internal interrupts (a concept that's particular to Ibex, it's not a RISC-V architecture concept): https://ibex-core.readthedocs.io/en/latest/03_reference/exception_interrupts.html#internal-interrupts

I think the WARL behaviour here is fine with regards to the spec as WLRL and WARL and more or less the same from the hardware's point of view:

> Implementations can return arbitrary bit patterns on the read of a WLRL field when the last write was of an illegal value, but the value returned should deterministically depend on the illegal written value and the value of the field prior to the write.

> Implementations can return any legal value on the read of a WARL field when the last write was of an illegal value, but the legal value returned should deterministically depend on the illegal written value and the architectural state of the hart

The only real difference is it's allowable to return an illegal value for a `WLRL` field when an illegal value is written.

The distinction matters more to software (where for WARL it can probe by writing arbitrary values to discover the legal ones for `WLRL` is should only ever write values it knows to be legal).

Reply:
Ok, looks good to me based on that table.